starknet_py/tests/e2e/client/fixtures/transactions.py

```diff
@@ -53,3 +60,73 @@ def block_with_deploy_account_number(
     """
     _, prepared_data = prepare_network
     return prepared_data.block_with_deploy_account_number
+
+
+@pytest_asyncio.fixture(scope="package")
+async def declare_v2_hello_starknet(gateway_account: Account) -> DeclareV2:
+    """
+    Returns DeclareV2 transaction.
+    """
+    hello_starknet_compiled = read_contract(
+        CONTRACTS_COMPILED_V1_DIR / "hello_starknet_compiled.json"
+    )
+    compiled_class_hash = compute_casm_class_hash(
+        create_casm_class(
+            read_contract(CONTRACTS_COMPILED_V1_DIR / "hello_starknet_compiled.casm")
+        )
+    )
+    return await gateway_account.sign_declare_v2_transaction(
+        hello_starknet_compiled,
+        compiled_class_hash=compiled_class_hash,
+        max_fee=MAX_FEE,
+    )
+
+
+@pytest_asyncio.fixture(scope="package")
+async def hello_starknet_class_hash_tx_hash(
+    gateway_client: Client, declare_v2_hello_starknet: DeclareV2
+) -> Tuple[int, int]:
+    """
+    Returns class_hash and transaction_hash from the declare_v2_hello_starknet transaction.
+    """
+    result = await gateway_client.declare(declare_v2_hello_starknet)
+    await gateway_client.wait_for_tx(
+        tx_hash=result.transaction_hash, wait_for_accept=True
+    )
+
+    return result.class_hash, result.transaction_hash
+
+
+@pytest_asyncio.fixture(scope="package")
+async def block_with_declare_v2_number(
+    hello_starknet_class_hash_tx_hash: Tuple[int, int], full_node_client
+) -> int:
+    """
+    Returns number of the block with DeclareV2 transaction
+    """
+    (_, tx_hash) = hello_starknet_class_hash_tx_hash
+    declare_v2_receipt = await full_node_client.get_transaction_receipt(tx_hash)
+    return declare_v2_receipt.block_number
+
+
+@pytest_asyncio.fixture(scope="package")
+async def replaced_class(account: Account, map_class_hash: int) -> Tuple[int, int, int]:
+    """
+    Returns block_number, contract_address and class_hash of transaction replacing implementation.
+    """
+    compiled_contract = read_contract("replace_class_compiled.json")
+
+    declare_result = await Contract.declare(account, compiled_contract, max_fee=MAX_FEE)
+    await declare_result.wait_for_acceptance()
+
+    deploy_result = await declare_result.deploy(max_fee=MAX_FEE)
+    await deploy_result.wait_for_acceptance()
+
+    contract = deploy_result.deployed_contract
+
+    resp = await contract.functions["replace_implementation"].invoke(
+        new_class=map_class_hash, max_fee=MAX_FEE
+    )
+    await resp.wait_for_acceptance()
+
+    return resp.block_number, contract.address, map_class_hash
```

Comment:
Are we testing anywhere the class replacement? Maybe let's add a test for the new `replace_class.cairo` to make sure it's working properly

Reply:
I don't think it's necessary. The test is working properly

Reply:
I thought about some test that would use the contract after replacing the class

Reply:
But I'm not sure if we should. That would not test the starknet.py feature, just cairo contract

Reply:
Checked it locally and it works 🎉